docs/DeveloperGuide.md

```diff
@@ -5,9 +5,262 @@
+the item list and will call `FileManager+saveData()` at the end of `Command+execute()`.
+> Commands that call `FileManager+saveData()` in `execute()`:
+> - `NewCommand`
+> - `UpdateCommand`
+> - `DeleteCommand`
+> - `AddCommand`
+> - `RemoveCommand`
+
+Item data is saved into a text file in `./data/items.txt` by generating a string for each `Item` in the list, containing the attributes of the `item`.
+The string generated will have the attributes in the format and order of `NAME,QUANTITY,PRICE,EXPIRY_DATE`. Since an `Item`'s expiry date is
+optional, `EXPIRY_DATE` can be the string `"no date"` if the expiry date is determined to not exist. This string will be written to the text file as mentioned.
+> For example, suppose the program wants to save the `Item` with attributes `name = "Apple"`, `quantity = 5`, `price = 2.50`, `expiryDate = 19-04-2024`
+> and the program uses `"|"` to indicate separation between groups.\
+> The program generates the string `"Apple|5|2.50|19-04-2024|"`
+
+This string is generated by the method `FileManager-getItemData()`
+
+#### Loading
+Loading is performed at the start of the program in `SuperTracker-run()` where it calls the method `FileManager+loadData()`.
+`loadData()` looks for the save text file at `./data/items.txt` and reads the lines from the file `items.txt` exists. As indicated earlier,
+each line containing a single `Item`'s data will be in the format and order of `NAME,QUANTITY,PRICE,EXPIRY_DATE`. Each attribute will be parsed into its
+relevant data type, creating a new `Item` with the extracted attributes, which is then added to the item list in `Inventory`. In the event where the line
+of data read is not in the correct format, or the attributes are unable to be parsed into its relevant data type (i.e. the string in the QUANTITY part reads "f1ve" instead of "5")_,
+the line will be ignored and a new `Item` will not be added to the list.
+
+The parsing of string data to an `Item` object is handled by the method `FileManager-parseItemData()`.
+
+### New Command
+The following is a class diagram of the NewCommand and its relevant dependencies<br>
+![NewCommandClass](uml-diagrams/NewCommandClass.png)
+
+The `NewCommand` class implements the `Command` interface and is responsible for handling the creation of new items in the `Inventory`.
+A NewCommand instance is created by the `parseNewCommand` method called by Parser, which ensures that the provided parameters (name, quantity, price, expiry date) are valid.
+
+#### Dependencies
+- `Item`: For creating the new item
+- `Inventory`: For adding the new item into the inventory
+- `Ui`: To notify the user about the successful execution of `NewCommand`
+- `FileManager`: To save the new item added onto the hard disk
+
+The following sequence diagram shows the execution of a NewCommand<br>
+![NewCommandSequence](uml-diagrams/NewCommandSequence.png)
+
+1. The `SuperTracker` class calls the `execute` method of `NewCommand`
```

Comment:
`SuperTracker` is not explained enough previously to let the reader understand it is responsible for a lot of the main functionality.